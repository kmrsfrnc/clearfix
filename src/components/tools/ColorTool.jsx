import React, { Component } from 'react';
import Color from 'color';

import InputValue from '../elements/InputValue';

import './ColorTool.scss';

class ColorTool extends Component {
  static getDerivedStateFromProps({ value }) {
    try {
      return {
        color: Color(value.trim()),
      };
    } catch (err) {
      return {
        color: null,
        mix: null,
      };
    }
  }

  state = {
    color: null,
  };

  render() {
    const { color } = this.state;

    if (!color) {
      return null;
    }

    return (
      <div>
        <h2>Color</h2>
        <div className="color-tool">
          <div className="color-tool__sample" style={{ backgroundColor: color.hex() }} />
          <div className="color-tool__values">
            <InputValue>
              {`rgb(${color.rgb().array().join(',')})`}
            </InputValue>
            <InputValue>
              {color.hsl().string()}
            </InputValue>
            <InputValue>
              {color.hex()}
            </InputValue>
          </div>
        </div>
      </div>
    );
  }
}

export default ColorTool;
