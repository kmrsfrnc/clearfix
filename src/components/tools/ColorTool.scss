.color-tool {
  display: flex;
}

.color-tool__values {
  padding-top: 0.5em;
}

.color-tool__sample {
  border-radius: 4px;
  margin-right: 1em;
  width: 8em;
}
