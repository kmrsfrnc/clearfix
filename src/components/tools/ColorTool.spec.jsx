import React from 'react';
import { shallow } from 'enzyme';

import ColorTool from './ColorTool';

describe('ColorTool', () => {
  it('parses color by name', () => {
    expect(shallow(<ColorTool value="whitesmoke" />)).toMatchSnapshot();
  });

  it('parses color by hex', () => {
    expect(shallow(<ColorTool value="#F5F5F5" />)).toMatchSnapshot();
  });

  it('parses color by rgb', () => {
    expect(shallow(<ColorTool value="rgb(245,245,245)" />)).toMatchSnapshot();
  });

  it('renders null if not a color', () => {
    expect(shallow(<ColorTool value="round" />).type()).toBeNull();
  });
});
