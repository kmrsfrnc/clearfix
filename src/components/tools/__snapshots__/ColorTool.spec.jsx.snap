// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorTool parses color by hex 1`] = `
<div>
  <h2>
    Color
  </h2>
  <div
    className="color-tool"
  >
    <div
      className="color-tool__sample"
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
    <div
      className="color-tool__values"
    >
      <InputValue>
        rgb(245,245,245)
      </InputValue>
      <InputValue>
        hsl(0, 0%, 96.1%)
      </InputValue>
      <InputValue>
        #F5F5F5
      </InputValue>
    </div>
  </div>
</div>
`;

exports[`ColorTool parses color by name 1`] = `
<div>
  <h2>
    Color
  </h2>
  <div
    className="color-tool"
  >
    <div
      className="color-tool__sample"
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
    <div
      className="color-tool__values"
    >
      <InputValue>
        rgb(245,245,245)
      </InputValue>
      <InputValue>
        hsl(0, 0%, 96.1%)
      </InputValue>
      <InputValue>
        #F5F5F5
      </InputValue>
    </div>
  </div>
</div>
`;

exports[`ColorTool parses color by rgb 1`] = `
<div>
  <h2>
    Color
  </h2>
  <div
    className="color-tool"
  >
    <div
      className="color-tool__sample"
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
    <div
      className="color-tool__values"
    >
      <InputValue>
        rgb(245,245,245)
      </InputValue>
      <InputValue>
        hsl(0, 0%, 96.1%)
      </InputValue>
      <InputValue>
        #F5F5F5
      </InputValue>
    </div>
  </div>
</div>
`;
